landoapi/api/landing_jobs.py

```diff
@@ -74,6 +74,22 @@ def put(landing_job_id: str, data: dict):
         )
 
 
+def try_convert_datetime(date: str) -> datetime:
```

Comment:
```suggestion
def datetime_from_string(date: str) -> datetime:
```

I think the word `try` is not necessary and could be slightly confusing.